src/Microsoft.DncEng.Configuration.Extensions/ServiceConfigurationExtensions.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+#if NETCOREAPP2_1
+using IHostEnvironment = Microsoft.Extensions.Hosting.IHostingEnvironment;
+#else
+using IHostEnvironment = Microsoft.Extensions.Hosting.IHostEnvironment;
+#endif
+
+namespace Microsoft.DncEng.Configuration.Extensions
+{
+    public static class ServiceConfigurationExtensions
+    {
+        public static IServiceCollection AddDefaultJsonConfiguration(this IServiceCollection services)
+        {
+            return services.AddSingleton(provider =>
+            {
+                IHostEnvironment env = provider.GetRequiredService<IHostEnvironment>();
+                IConfiguration config = new ConfigurationBuilder()
+                    .AddDefaultJsonConfiguration(env)
+                    .Build();
+                return config;
+            });
+        }
+
+        /// <summary>
+        ///   Adds the "Default" json file configuration to the <see cref="IConfigurationBuilder"/>.
+        ///   This loads the files '.config/settings.json' and '.config/settings.&lt;Environment&gt;.json', and maps Key Vault secrets and Azure App Configuration keys into the configuration.
+        ///   Key Vault:
+        ///     Key Vault secrets are read from the Key Vault referenced by the '<see cref="ConfigurationConstants.KeyVaultUriConfigurationKey"/>' configuration value and used to replace any references of the form '[vault(my_special_secret_name)]' with the corresponding secret value
+        ///   App Configuration:
+        ///     App Configuration keys are retrieved from the '<see cref="ConfigurationConstants.AppConfigurationUriConfigurationKey"/>' configuration value and replace references matching '[config(my_app_config_key)]' with the key value
+        ///
+        ///   Authentication is handled by either MSI (optionally using the '<see cref="ConfigurationConstants.ManagedIdentityIdConfigurationKey"/>' configuration value for a user-assigned managed identity), or VS/az cli authentication.
+        ///   Values will be refreshed every '<see cref="ConfigurationConstants.ReloadTimeSecondsConfigurationKey"/>' seconds.
+        /// </summary>
+        /// <param name="builder"></param>
```

Comment:
Empty param comments just waste space.  I'd just delete the `<param>` and `<returns>` elements

Reply:
These were auto-added. I will remove them.